azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -406,28 +494,61 @@ func resourceArmEventGridEventSubscriptionDelete(d *schema.ResourceData, meta in
 	return nil
 }
 
+func expandEventGridExpirationTime(d *schema.ResourceData) (*date.Time, error) {
+	if expirationTimeUtc, ok := d.GetOk("expiration_time_utc"); ok {
+		if expirationTimeUtc == "" {
+			return nil, nil
+		}
+
+		parsedExpirationTimeUtc, err := date.ParseTime(time.RFC3339, expirationTimeUtc.(string))
+		if err != nil {
+			return nil, err
+		}
+
+		return &date.Time{Time: parsedExpirationTimeUtc}, nil
+	}
+
+	return nil, nil
+}
+
 func expandEventGridEventSubscriptionDestination(d *schema.ResourceData) eventgrid.BasicEventSubscriptionDestination {
-	if _, ok := d.GetOk("storage_queue_endpoint"); ok {
-		return expandEventGridEventSubscriptionStorageQueueEndpoint(d)
+	if _, ok := d.GetOk(AzureFunctionEndpointID); ok {
+		return expandEventGridEventSubscriptionAzureFunctionEndpointID(d)
 	}
 
-	if _, ok := d.GetOk("eventhub_endpoint"); ok {
+	if _, ok := d.GetOk(EventHubEndpointID); ok {
+		return expandEventGridEventSubscriptionEventHubEndpointID(d)
+	} else if _, ok := d.GetOk(EventHubEndpoint); ok {
```

Comment:
and similarly for these we can
```suggestion
	if v, ok := d.GetOk(EventHubEndpointID); ok {
		return &eventgrid.EventHubEventSubscriptionDestination{
			EndpointType: eventgrid.EndpointTypeEventHub,
			EventHubEventSubscriptionDestinationProperties: &eventgrid.EventHubEventSubscriptionDestinationProperties{
				ResourceID: utils.String(v.(string)),
			},)
	} else if _, ok := d.GetOk(EventHubEndpoint); ok {
```